numpy/f2py/src/fortranobject.c

```diff
@@ -590,6 +590,11 @@ static void f2py_report_on_array_copy_fromany(void) {
  * $Id: fortranobject.c,v 1.52 2005/07/11 07:44:20 pearu Exp $
  */
 
+static int check_and_fix_dimensions(const PyArrayObject* arr,
+                                    const int rank,
+                                    npy_intp *dims,
+                                    char *msgbuffer, int msgbuffer_len);
```

Comment:
Why are these passed in, and not just declared locally? Does anything else even use them?

Seems like there could just be `char msgbuffer[200]` inside the `if (size != arr_size)  }{ ... }` further down.